compiler-plugin/src/main/java/io/ballerina/stdlib/email/compiler/EmailServiceValidator.java

```diff
@@ -276,6 +279,22 @@ private void validateParameter(SeparatedNodeList<ParameterNode> parameterNodes,
         }
     }
 
+    private boolean validateOnMessageParams(FunctionDefinitionNode functionDefinitionNode) {
+        FunctionTypeSymbol functionTypeSymbol = ((MethodSymbol) ctx.semanticModel().symbol(functionDefinitionNode)
+                .get()).typeDescriptor();
+        List<ParameterSymbol> inputParams;
+        if (functionTypeSymbol.params().isPresent()) {
+            inputParams = functionTypeSymbol.params().get();
+            if (inputParams.size() != 1) {
+                return false;
+            }
+            if (inputParams.get(0).typeDescriptor().getName().isPresent()) {
+                return EMAIL_MESSAGE.equals(inputParams.get(0).typeDescriptor().getName().get());
```

Comment:
I still don't get the idea. Is any parameter type named `Message` allowed as the input parameter? Don't we have to limit this to the `email` module-specific type (`email:Message`)?

Reply:
We have created an issue to identify similar issues https://github.com/ballerina-platform/ballerina-standard-library/issues/2635

Reply:
Added validation to `email` package as well.